langchain/output_parsers/smart_boolean.py

```diff
@@ -0,0 +1,44 @@
+from langchain.output_parsers.boolean import BooleanOutputParser
+
+
+class SmartBooleanOutputParser(BooleanOutputParser):
+    """Boolean output parser that can handle changes in capitalization, whitespace,
+    punctuation, and additional words in the output.
+
+    Works by splitting the output of the language model into words and checking if any of them
+    match the true or false values. If multiple words match, the first one is used.
+
+    This output parser is useful when your LLM's give responses like: "Yes, consider 4 reasons
+    why...". Since `BooleanOutputParser` only checks for exact matches, it would fail on this
+    input. (However if you only want to check for exact matches, `BooleanOutputParser` is what
+    you probabbly want.)
+    """
+
+    true_val: str = "YES"
+    false_val: str = "NO"
+
+    def parse(self, text: str) -> bool:
+        """Parse the output of an LLM call to a boolean.
+
+        Args:
+            text: output of language model
+
+        Returns:
+            boolean
+
+        """
+        for word in text.split():
+            word = word.strip("\r\n\t \"',.:;!?[]{}|/\\").lower()
```

Comment:
wonder if using re would be a bit more robust + succinct (replace yes/no with true_val/false_val)

![Screenshot 2023-05-11 at 8 35 29 PM](https://github.com/hwchase17/langchain/assets/130488702/2e2b8c9d-9076-4924-9416-be158a2e7143)